File: SafetyProgram.UI.DocumentObject/ChemicalTableUI/RibbonTabs/IChemicalTableRibbonViewModel.cs
using System.Collections.ObjectModel;
using System.ComponentModel;
using SafetyProgram.Core.Commands;
using SafetyProgram.Core.Models;

namespace SafetyProgram.UI.DocumentObject.ChemicalTableUI
{
    /// <summary>
    /// Defines a ViewModel for a cheimcal table ribbon view.
    /// </summary>
    internal interface IChemicalTableRibbonViewModel : INotifyPropertyChanged
    {
        /// <summary>
        /// Get or set the current search (for searching chemicals)
        /// </summary>
        string Search { get; set; }

        /// <summary>
        /// Get the search result yielded from the search string.
        /// </summary>
        ObservableCollection<IChemical> SearchResult { get; }

        /// <summary>
        /// Get a group of commands that act on the chemicaltable.
        /// </summary>
        IChemicalTableCommands Commands { get; }
    }
}

File: SafetyProgram.UI.ViewModels/IChemicalTableRibbonViewModel.fs
namespace SafetyProgram.UI.ViewModels

open SafetyProgram.UI.Models
open System.Collections.ObjectModel
open System.ComponentModel

type IChemicalTableRibbonViewModel = 
    
    inherit INotifyPropertyChanged

    abstract Search : string with get, set
    abstract SearchResult : ObservableCollection<string> with get